src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -424,4 +434,44 @@ public class GenerateProtoTask extends DefaultTask {
     }
   }
 
+  static List<String> generateCmds(String baseCmd, List<File> protoFiles, int cmdLengthLimit) {
+    List<String> cmds = []
+    StringBuilder currentCmd = new StringBuilder(baseCmd)
+    for (File proto: protoFiles) {
+      String protoFileName = proto
+      // Check if appending the next proto string will overflow the cmd length limit
+      if (currentCmd.length() + protoFileName.length() + 1 > cmdLengthLimit) {
+        // Add the current cmd before overflow
+        cmds.add(currentCmd.toString())
+        currentCmd.setLength(baseCmd.length())
+      }
+      // Append the proto file to the command
+      currentCmd.append(" ").append(protoFileName)
+    }
+    // Add the last cmd for execution
+    cmds.add(currentCmd.toString())
+    return cmds
+  }
+
+  private static int getCmdLengthLimit() {
```

Comment:
Please also split out the actual handling logic, something like `getCmdLengthLimit(os, version)`, and add a test for it.